### ISSUE TITLE
Inventory System & Items

### ISSUE THREAD

**Issue by Lithrun** [AoTTG-2/AoTTG-2#114]
**Is your feature request related to a problem? Please describe.**
Players will have an inventory system with which they can select items and use them. They can access these items via an item wheel, whose design is identical to the interaction wheel. Players can assign items to their custom character and use those to their advantage during the game.

**Describe the solution you'd like**
An inventory system which has items which are easy to extend upon

- [x] Develop an inventory system
- [x] Flare Guns. A player always has a flare gun in its inventory
- [ ] Lantern. Allows the player to see better at night
- [ ] Rifle. A rifle can be equipped, which has a limited amount of ammunition and a long reload time, but it is able to temporarily blind a titan by shooting it in the eyes. It cannot be used while ODMing.

Once this task is done, additional items can be added via new feature requests.

**Comment by Lithrun** [comment]
Issue on hold until #33

**Comment by Prolonpo** [comment]
From @IRI,
• include torches in this inventory system which would allow players to see better at night, albeit at a shorter radius than lantern but will have the advantage of not being able to be easily broken compared to lanterns.
• Include bow and arrow which will allow players to blind titans. Shorter range than rifles, but faster reload than rifles.

Summary post: https://discord.com/channels/681641241125060652/683107211983323164/780831072078135367

**Comment by Prolonpo** [comment]
From @BiawA,

Flares should have the option of being limited and unlimited.

Summary post: https://discord.com/channels/681641241125060652/683107211983323164/780979242070441984

**Comment by Prolonpo** [comment]
Ability to loot gas and blades from dead corpses - this idea has being proposed multiple times (not too sure who was the original poster of this, so I'll leave this blank for now)

Reference post: https://discord.com/channels/681641241125060652/683107211983323164/787890007402610698

**Comment by Prolonpo** [comment]
From @antigasp, items which are very heavy and that would reduce a player's flying speed should reduce a player's ACL (acceleration) points to a certain level.

For example, carrying resupply gears would reduce all characters' ACL to 100 regardless of whether the player has ACL of 125 or Levi's ACL.

This can lead to the creation of characters that can be considered pack mules - slow generally, but not penalized when carrying heavy items.

Reference: https://discord.com/channels/681641241125060652/683107211983323164/804586240711458816

**Comment by Adonca2203** [comment]
Currently the inventory system works like a charm. You are able to pick up items that are placed on the map and load a flexible radial menu which is populated with your items and refreshed upon item addition or removal. You are also able to use any item in your inventory wheel by clicking on it. I'm currently working on adding functionally for the lantern and then I'll move onto the rifle.

**Comment by Adonca2203** [comment]
There's a bug with the light on the lantern in which whenever i face south on any map, the light distance shortens itself on the ground to a point where it's near invisible however it still lights up walls. Unsure what the cause is, i tried changing the ground texture for a wall texture and it did not fix the issue. For now I will be moving onto the rilfe.
